pkg/services/datasources/service/datasource_service.go

```diff
@@ -183,12 +177,21 @@ func (s *Service) AddDataSource(ctx context.Context, cmd *models.AddDataSourceCo
 }
 
 func (s *Service) DeleteDataSource(ctx context.Context, cmd *models.DeleteDataSourceCommand) error {
+	err := s.SecretsStore.Del(ctx, cmd.OrgID, cmd.Name, "datasource")
+	if err != nil {
+		return err
+	}
 	return s.SQLStore.DeleteDataSource(ctx, cmd)
```

Comment:
Might be worth deleting the secret after datasource has been deleted

Reply:
Yup I noticed that during the demo.

Reply:
Done on 145cac0